src/main/kotlin/org/rust/ide/inspections/fixes/AttachFileToModuleFix.kt

```diff
@@ -0,0 +1,187 @@
+        val rsFile = startElement as? RsFile ?: return
+        val availableModules = findAvailableModulesForFile(project, rsFile)
+        if (availableModules.isEmpty()) return
+
+        if (availableModules.size == 1) {
+            insertFileToModule(rsFile, availableModules[0])
+        } else if (availableModules.size > 1) {
+            selectModule(rsFile, availableModules)?.let { insertFileToModule(rsFile, it) }
+        }
+    }
+
+    override fun startInWriteAction(): Boolean {
+        return false
+    }
+
+    companion object {
+        fun findAvailableModulesForFile(project: Project, file: RsFile): List<RsMod> {
+            val virtualFile = file.virtualFile ?: return emptyList()
+            val pkg = project.cargoProjects.findPackageForFile(virtualFile) ?: return emptyList()
+
+            val directory = virtualFile.parent ?: return emptyList()
+            val modules = mutableListOf<RsMod>()
+
+            if (file.isModuleFile) {
+                // package target roots in parent directory
+                for (target in pkg.targets) {
+                    val crateRoot = target.crateRoot ?: continue
+                    if (crateRoot.parent == directory.parent) {
+                        modules.addIfNotNull(crateRoot.toPsiFile(project)?.rustFile)
+                    }
+                }
+            } else {
+                // mod.rs in the same directory
+                modules.addIfNotNull(findModule(file, project, directory.findFileByRelativePath(RsConstants.MOD_RS_FILE)))
+
+                // module file in parent directory
+                if (pkg.edition == CargoWorkspace.Edition.EDITION_2018) {
+                    val parent = directory.parent
+                    modules.addIfNotNull(findModule(file, project, parent?.findFileByRelativePath("${directory.name}.rs")))
+                }
+
+                // package target roots in the same directory
+                for (target in pkg.targets) {
+                    val crateRoot = target.crateRoot ?: continue
+                    if (crateRoot.parent == directory) {
+                        modules.addIfNotNull(crateRoot.toPsiFile(project)?.rustFile)
+                    }
+                }
+            }
+
+            return modules
+        }
+    }
+}
+
+private fun selectModule(file: RsFile, availableModules: List<RsMod>): RsMod? {
+    if (isUnitTestMode) return MOCK!!(file, availableModules)
+
+    val box = ComboBox<RsMod>()
+    with(box) {
+        for (module in availableModules) {
+            addItem(module)
+        }
+        renderer = SimpleListCellRenderer.create("") {
+            val root = it.containingCargoPackage?.rootDirectory
+            val path = it.containingFile.virtualFile.pathAsPath
+            (root?.relativize(path) ?: path).toString()
+        }
+    }
+
+    val dialog = dialog("Select a module", panel {
+        row { box() }
```

Comment:
```suggestion
        row { box(CCFlags.growX) }
```

I think 
<img width="249" alt="Screen Shot 2020-06-08 at 10 38 19" src="https://user-images.githubusercontent.com/2539310/84004587-7015d180-a974-11ea-94cf-3861909c76db.png">
is better than

<img width="246" alt="Screen Shot 2020-06-08 at 10 35 35" src="https://user-images.githubusercontent.com/2539310/84004606-77d57600-a974-11ea-9558-436f77f49c10.png">